src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -5202,33 +5215,40 @@ private ModelRecord handleUnexpectedFailedCacheEntry(CacheEntry<?> ce, ModelReco
         if (failure == null) {
             return mr; // safeguard timeout case (didn't see load fail but timed out waiting for it)
         }
-        if (failure instanceof ModelLoadException
-            && ((ModelLoadException) failure).getTimeout() == KVSTORE_LOAD_FAILURE) {
-            long failureAge = currentTimeMillis() - ce.loadCompleteTimestamp;
-            if (failureAge > 30_000 && failureAge > 30_000
-                    + ThreadLocalRandom.current().nextLong(30_000)) { // Randomize to avoid thunder
-                ModelRecord newMr = registry.get(ce.modelId);
-                if (newMr == null ? mr != null : (mr == null || newMr.getVersion() == mr.getVersion())) {
-                    // First replace the entry with a later-expiring one to block concurrent attempts
-                    CacheEntry<?> replacement = new CacheEntry<>(ce);
-                    if (runtimeCache.replaceQuietly(ce.modelId, ce, replacement)) {
-                        ce.remove();
-                        ce = replacement;
+        if (!(failure instanceof ModelLoadException)
+            || ((ModelLoadException) failure).getTimeout() != KVSTORE_LOAD_FAILURE) {
+            // We assume that this is an expired entry yet to be cleaned up
+            if (ce.remove()) {
+                logger.info("Removed kv-store failure cache entry for model " + ce.modelId);
+            }
+            return mr;
+        }
+
+        long failureAge = currentTimeMillis() - ce.loadCompleteTimestamp;
+        if (failureAge > 30_000 && failureAge > 30_000
```

Comment:
can the first `failureAge > 30_000` be removed from this if-condition? ... since the first part will always be true if the second part of the condition is true

Reply:
It's actually a small optimization.. so that we don't generate the random number most of the time (the `&&` will short-circuit if the first condition is false.

Reply:
ah, neat, cool